### PR TITLE
add statsd support

### PR DESCRIPTION
**Pull request by svenwltr**
@rebuy-de/prp-kubernetes-deployment Please review.

### CODE REVIEW

File: pkg/api/generate.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ func (app *App) Generate(project, branchName string) ([]runtime.Object, error) {
 		"Branch":  branchName,
 	}).Infof("deploying project")
 
+	app.Clients.Statsd.Increment("deployments")
```
Comment (bjoernhaeuser) [inline pkg/api/generate.go]: no labels? :(

Comment (svenwltr) [inline pkg/api/generate.go]: We would have to configure the exporter: https://github.com/prometheus/statsd_exporter#metric-mapping-and-configuration

Should we do?

Comment (svenwltr) [inline pkg/api/generate.go]: Also its prefixed by `kubernetes-deployment`.

**Comment by svenwltr** [comment]
I moved the `Increment` to apply (since this makes more sense) and added the project and branch name. We have to configure the statsd exporter for create proper Prometheus labels.

**Comment by svenwltr** [comment]
Hmm ... https://github.com/prometheus/statsd_exporter#dogstatsd-extensions

**Comment by svenwltr** [comment]
I added the "dogstatsd" tags. It look like this:

```
$ nc -l -u 8125
kubernetes-deployment.github.rate.remaining:4968|g
kubernetes-deployment.generate:1|c|#project:blue-web,branch:master
kubernetes-deployment.github.rate.remaining:4968|g
kubernetes-deployment.github.rate.remaining:4966|g
kubernetes-deployment.github.rate.remaining:4965|g
kubernetes-deployment.github.rate.remaining:4964|g
```

@rebuy-de/prp-kubernetes-deployment Please review.

**Pull request merged**